gprofiler/utils/fs.py

```diff
@@ -15,3 +18,31 @@ def safe_copy(src: str, dst: str) -> None:
     dst_tmp = f"{dst}.tmp"
     shutil.copy(src, dst_tmp)
     os.rename(dst_tmp, dst)
+
+
+def is_rw_exec_dir(path: str) -> bool:
+    """
+    Is 'path' rw and exec?
+    """
+    test_script = Path(path) / "t.sh"
+
+    # try creating & writing
+    try:
+        os.makedirs(path, 0o755, exist_ok=True)
+        test_script.write_text("#!/bin/sh\nexit 0")
+        test_script.chmod(0o755)
+    except OSError as e:
+        if e.errno == errno.EROFS:
```

Comment:
Here I'm checking for RO because that's what I expect and try to avoid. Other errors are unrelated and signal another type of error in our environment / code, so I don't want to swallow them

Reply:
Hmm I need to raise them, yeah

Reply:
I mean you swallow other errors and potentially not removing `t.sh` (if the error is with `chmod()`)